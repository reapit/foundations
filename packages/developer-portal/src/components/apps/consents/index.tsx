import React, { Dispatch, FC, SetStateAction, useEffect, useState } from 'react'
import { Button, ButtonGroup, elMb11, Loader, PersistentNotification, Table, Title } from '@reapit/elements'
import { useAppState } from '../state/use-app-state'
import { useParams } from 'react-router-dom'
import { handleSetAppId } from '../utils/handle-set-app-id'
import { SendFunction, useReapitGet, useReapitUpdate } from '@reapit/use-reapit-data'
import { reapitConnectBrowserSession } from '../../../core/connect-session'
import { GetActionNames, getActions, UpdateActionNames, updateActions } from '@reapit/use-reapit-data'
import { useGlobalState } from '../../../core/use-global-state'
import { Marketplace } from '@reapit/foundations-ts-definitions'
import dayjs from 'dayjs'

export const handleResendEmail =
  (
    resendEmail: SendFunction<Marketplace.ResendAppRevisionConsentModel, boolean>,
    setConsentId: Dispatch<SetStateAction<string | null>>,
    appConsentsRefresh: () => void,
    consentId: string | null,
    developerEmail?: string,
  ) =>
  () => {
    const sentEmail = async () => {
      const response = await resendEmail({ actionedBy: developerEmail })
      setConsentId(null)

      if (response) {
        appConsentsRefresh()
      }
    }

    if (consentId) {
      sentEmail().catch((error) => console.error(error))
    }
  }

export const handleSetConsentId = (setConsentId: Dispatch<SetStateAction<string | null>>, consentId?: string) => () => {
  if (consentId) {
    setConsentId(consentId)
  }
}

export const handleSendConstents =
  (
    createConsentEmails: SendFunction<Marketplace.CreateAppRevisionConsentsModel, boolean>,
    appConsentsRefresh: () => void,
    developerEmail?: string,
  ) =>
  async () => {
    const response = await createConsentEmails({ actionedBy: developerEmail })

    if (response) {
      appConsentsRefresh()
    }
  }

export const AppConsentsPage: FC = () => {
  const { globalDataState } = useGlobalState()
  const [consentId, setConsentId] = useState<string | null>(null)
  const { appsDataState, setAppId } = useAppState()
  const { currentDeveloper } = globalDataState
  const developerId = currentDeveloper?.id
  const developerEmail = currentDeveloper?.email
  const { appId } = useParams<'appId'>()

  useEffect(handleSetAppId(setAppId, appId), [appId])

  const { appDetail, appRevisions } = appsDataState
  const { name } = appDetail ?? {}
  const latestRevision = appRevisions?.data ? appRevisions.data[0] : null

  const [appConsents, appConsentsLoading, , appConsentsRefresh] = useReapitGet<Marketplace.AppRevisionConsentModel[]>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getRevisionConsents],
    uriParams: {
      appId,
      revisionId: latestRevision?.id,
    },
    fetchWhenTrue: [appId, latestRevision, appDetail?.pendingRevisions, appDetail?.isListed],
  })

  const [, , createConsentEmails] = useReapitUpdate<Marketplace.CreateAppRevisionConsentsModel, boolean>({
    reapitConnectBrowserSession,
    action: updateActions[UpdateActionNames.createConsentEmails],
    method: 'POST',
    uriParams: {
      appId,
      revisionId: latestRevision?.id,
    },
  })

  const [installations] = useReapitGet<Marketplace.InstallationModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getInstallations],
    queryParams: {
      appId,
      pageNumber: 1,
      pageSize: 999,
      isInstalled: true,
      developerId,
    },
    fetchWhenTrue: [developerId],
  })

  const [, , resendEmail] = useReapitUpdate<Marketplace.ResendAppRevisionConsentModel, boolean>({
    reapitConnectBrowserSession,
    action: updateActions[UpdateActionNames.resendConsentEmail],
    method: 'POST',
    uriParams: {
      appId,
      revisionId: latestRevision?.id,
      consentId,
    },
  })

  useEffect(handleResendEmail(resendEmail, setConsentId, appConsentsRefresh, consentId, developerEmail), [consentId])

  return (
    <>
      {appConsentsLoading || !appConsents ? (
        <Loader />
      ) : appConsents.length ? (
        <>
          <Title>{name} Consents</Title>
          <Table
            numberColumns={6}
            rows={appConsents.map((consent) => ({
              cells: [
                {
                  label: 'Client',
                  value: installations?.data?.find((installation) => installation.id === consent.installationId)
                    ?.customerName,
                  icon: 'property',
                  cellHasDarkText: true,
                  narrowTable: {
                    showLabel: true,
                  },
                },
                {
                  label: 'Email',
                  value: consent.installedBy,
                  narrowTable: {
                    showLabel: true,
                  },
                },
                {
                  label: 'Date Sent',
                  value: dayjs(consent.sentDate).format('DD/MM/YYYY HH:mm'),
                  narrowTable: {
                    showLabel: true,
                  },
                },
                {
                  label: 'Status',
                  value: `${consent.status?.charAt(0).toUpperCase()}${consent.status?.slice(1)}`,
                  narrowTable: {
                    showLabel: true,
                  },
                },
                {
                  label: 'Date Accepted',
                  value: consent.approvalDate ? dayjs(consent.approvalDate).format('DD/MM/YYYY HH:mm') : '-',
                  narrowTable: {
                    showLabel: true,
                  },
                },
              ],
              ctaContent: {
                headerContent: 'Resend Email',
                icon: 'email',
                onClick: handleSetConsentId(setConsentId, consent.id),
              },
            }))}
          />
        </>
      ) : (
        <>
          <Title>{name} Consents</Title>
          <PersistentNotification className={elMb11} intent="primary" isExpanded isFullWidth isInline>
            No record of any consents for this app - it looks like you need to send an email requesting that your
            customers agree to your new permission settings. You can do this below or our team will do it as part of the
            review process.
          </PersistentNotification>
          <ButtonGroup alignment="left">
            <Button
              fixedWidth
              onClick={handleSendConstents(createConsentEmails, appConsentsRefresh, developerEmail)}
              intent="primary"
            >
              Send Requests
            </Button>
          </ButtonGroup>
        </>
      )}
    </>
  )
}

export default AppConsentsPage
