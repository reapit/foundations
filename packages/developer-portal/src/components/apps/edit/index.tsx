import React, { ChangeEvent, FC, useEffect } from 'react'
import {
  Button,
  ButtonGroup,
  Loader,
  MobileControls,
  PersistentNotification,
  Tabs,
  TabsOption,
  Title,
  useMediaQuery,
  useModal,
} from '@reapit/elements'
import { useAppState } from '../state/use-app-state'
import { handleSetAppId } from '../utils/handle-set-app-id'
import { NavigateFunction, useLocation, useNavigate, useParams } from 'react-router-dom'
import { AppEditTab } from './edit-page-tabs'
import { AppEditForm } from './app-edit-form'
import { Helper } from '../page/helper'
import Routes from '../../../constants/routes'

export const handleChangeTab =
  (navigate: NavigateFunction, appId?: string) => (event: ChangeEvent<HTMLInputElement>) => {
    if (appId) {
      navigate(`${Routes.APPS}/${appId}/edit/${event.target.value}`)
    }
  }

export const AppEditPage: FC = () => {
  const navigate = useNavigate()
  const location = useLocation()
  const { appId } = useParams<'appId'>()
  const { appsDataState, setAppId } = useAppState()
  const { isMobile } = useMediaQuery()
  const { Modal, openModal, closeModal } = useModal()

  useEffect(handleSetAppId(setAppId, appId), [appId])

  const { appDetail, appDetailLoading } = appsDataState
  const { pathname } = location

  return appDetailLoading ? (
    <Loader />
  ) : appDetail ? (
    <>
      <Title>{appDetail.name}</Title>
      {isMobile && (
        <Modal title="Controls">
          <Helper />
          <ButtonGroup alignment="right">
            <Button intent="default" onClick={closeModal}>
              Close
            </Button>
          </ButtonGroup>
        </Modal>
      )}
      <Tabs
        isFullWidth
        name="app-edit-tabs"
        onChange={handleChangeTab(navigate, appId)}
        options={
          [
            {
              id: AppEditTab.general,
              value: AppEditTab.general,
              text: 'About Listings',
              isChecked:
                pathname.includes(AppEditTab.general) || pathname === Routes.APPS_EDIT.replace(':appId', appId ?? ''),
            },
            {
              id: AppEditTab.appListing,
              value: AppEditTab.appListing,
              text: 'AppMarket Listing',
              isChecked: pathname.includes(AppEditTab.appListing),
            },
            appDetail.authFlow === 'authorisationCode' && {
              id: AppEditTab.authentication,
              value: AppEditTab.authentication,
              text: 'Authentication',
              isChecked: pathname.includes(AppEditTab.authentication),
            },
            {
              id: AppEditTab.permissions,
              value: AppEditTab.permissions,
              text: 'Permissions',
              isChecked: pathname.includes(AppEditTab.permissions),
            },
            appDetail.authFlow === 'authorisationCode' && {
              id: AppEditTab.acIntegration,
              value: AppEditTab.acIntegration,
              text: 'AgencyCloud Integration',
              isChecked: pathname.includes(AppEditTab.acIntegration),
            },
          ].filter(Boolean) as TabsOption[]
        }
      />
      <AppEditForm />
      <MobileControls onClick={openModal} />
    </>
  ) : (
    <PersistentNotification intent="primary" isExpanded isFullWidth isInline>
      No record of this app found
    </PersistentNotification>
  )
}

export default AppEditPage
