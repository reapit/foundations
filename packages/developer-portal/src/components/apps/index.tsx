import { PusherProvider } from '@harelpls/use-pusher'
import { useReapitConnect } from '@reapit/connect-session'
import React, { FC } from 'react'
import { URLS } from '../../constants/api'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { AppsPage } from './page'
import { AppProvider } from './state/use-app-state'

export const Apps: FC = () => {
  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)

  if (!connectSession) return null

  return (
    <PusherProvider
      cluster="eu"
      clientKey={process.env.PUSHER_KEY}
      authEndpoint={`${URLS.DEPLOYMENT_SERVICE_HOST}pusher/auth`}
      auth={{
        headers: {
          Authorization: connectSession.idToken,
        },
      }}
    >
      <AppProvider>
        <AppsPage />
      </AppProvider>
    </PusherProvider>
  )
}

export default Apps
