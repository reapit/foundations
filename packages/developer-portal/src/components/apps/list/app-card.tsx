import React, { FC } from 'react'
import { useNavigate } from 'react-router'
import { Marketplace } from '@reapit/foundations-ts-definitions'
import Routes from '../../../constants/routes'
import { Card, PlaceholderImage, elFadeIn } from '@reapit/elements'
import { navigateRoute } from '../../../utils/navigation'
import { cx } from '@linaria/core'
import { cardCursor } from './__styles__'

export interface AppCardProps {
  app: Marketplace.AppSummaryModel
}

export const AppCard: FC<AppCardProps> = ({ app }) => {
  const navigate = useNavigate()

  const { id, name, isDirectApi, developer, iconUri } = app

  return (
    <Card
      className={cx(elFadeIn, cardCursor)}
      onClick={navigateRoute(navigate, `${Routes.APPS}/${id}`)}
      hasMainCard
      mainCardHeading={name}
      mainCardSubHeading={developer}
      mainCardSubHeadingAdditional={isDirectApi ? 'Integration' : ''}
      mainCardAvatarUrl={iconUri ?? <PlaceholderImage placeholder="placeholderSmall" size={50} />}
    />
  )
}
