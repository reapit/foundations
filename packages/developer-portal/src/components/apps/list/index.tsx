import React, { FC } from 'react'
import {
  Button,
  ButtonGroup,
  Col,
  Title,
  useMediaQuery,
  Grid,
  useModal,
  Pagination,
  elMb11,
  MobileControls,
} from '@reapit/elements'
import { AppCard } from './app-card'
import { useAppState } from '../state/use-app-state'
import { Helper } from '../page/helper'

export const AppsList: FC = () => {
  const { isMobile } = useMediaQuery()
  const { appsDataState } = useAppState()
  const { Modal, openModal, closeModal } = useModal()

  const { apps, appsSetPageNumber } = appsDataState
  const pageNumber = apps?.pageNumber ?? 1
  const totalCount = apps?.totalCount ?? 0
  const pageSize = apps?.pageSize ?? 0
  const numberPages = Math.ceil((totalCount ?? 1) / (pageSize ?? 1))
  return (
    <>
      <Title>My Apps</Title>
      {isMobile && (
        <Modal title="Controls">
          <Helper />
          <ButtonGroup alignment="right">
            <Button intent="default" onClick={closeModal}>
              Close
            </Button>
          </ButtonGroup>
        </Modal>
      )}
      <Grid className={elMb11}>
        {apps?.data?.map((app) => (
          <Col key={app.id}>
            <AppCard app={app} />
          </Col>
        ))}
      </Grid>
      {numberPages > 1 && (
        <Pagination callback={appsSetPageNumber} currentPage={pageNumber} numberPages={numberPages} />
      )}
      <MobileControls onClick={openModal} />
    </>
  )
}

export default AppsList
