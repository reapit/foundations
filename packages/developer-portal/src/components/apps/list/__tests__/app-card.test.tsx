import React from 'react'
import { render } from '../../../../tests/react-testing'
import { AppCard } from '../app-card'
import { mockAppDetailModel } from '../../../../tests/__stubs__/apps'
import { useReapitConnect } from '@reapit/connect-session'

jest.mock('../../state/use-app-state')
jest.mock('@reapit/connect-session', () => ({
  ReapitConnectBrowserSession: jest.fn(),
  useReapitConnect: jest.fn(() => ({
    connectSession: {
      loginIdentity: {
        groups: [],
      },
    },
  })),
}))

const mockUseReapitConnect = useReapitConnect as jest.Mock

describe('AppCard', () => {
  it('should match a snapshot for a regular user', () => {
    expect(render(<AppCard app={mockAppDetailModel} />)).toMatchSnapshot()
  })

  it('should match a snapshot for an admin', () => {
    mockUseReapitConnect.mockReturnValueOnce({
      connectSession: {
        loginIdentity: {
          groups: ['FoundationsDeveloperAdmin'],
        },
      },
    })
    expect(render(<AppCard app={mockAppDetailModel} />)).toMatchSnapshot()
  })

  it('should match a snapshot for an admin with delete protection', () => {
    mockUseReapitConnect.mockReturnValueOnce({
      connectSession: {
        loginIdentity: {
          groups: ['FoundationsDeveloperAdmin'],
        },
      },
    })
    expect(render(<AppCard app={{ ...mockAppDetailModel, deletionProtection: true }} />)).toMatchSnapshot()
  })
})
