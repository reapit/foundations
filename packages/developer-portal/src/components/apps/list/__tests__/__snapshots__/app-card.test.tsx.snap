// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCard should match a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ecbqyq6-el-snack-holder"
      />
      <div
        class="e6p05ya-el-fade-in cbaewf6-card-cursor ertn4py-el-card-wrap"
      >
        <div
          class="ec7f1kt-el-card-context-menu-wrapper"
        >
          <div
            class="ewpy0cr-el-card-context-menu-toggle"
          >
            <span
              class="ej9yrfo-el-icon"
              style="font-size: 1.25rem;"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="19"
                  cy="4"
                  fill="currentColor"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="10"
                  fill="currentColor"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="16"
                  fill="currentColor"
                  r="1"
                />
              </svg>
            </span>
          </div>
          <div
            class="eh85zp4-el-card-context-menu-items"
          >
            <div
              class="euknxbk-el-card-context-menu-item"
            >
              <span
                class="ej9yrfo-el-icon"
                style="font-size: 1.25rem;"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M15.8336 17.2478C16.1265 17.5407 16.6014 17.5407 16.8943 17.2478C17.1872 16.955 17.1872 16.4801 16.8943 16.1872L10.7071 10L16.8943 3.81282C17.1872 3.51992 17.1872 3.04505 16.8943 2.75216C16.6014 2.45926 16.1265 2.45926 15.8336 2.75216L9.64643 8.93934L3.8128 3.10571C3.51991 2.81282 3.04503 2.81282 2.75214 3.10571C2.45925 3.3986 2.45925 3.87348 2.75214 4.16637L8.58577 10L2.75214 15.8336C2.45925 16.1265 2.45925 16.6014 2.75214 16.8943C3.04503 17.1872 3.51991 17.1872 3.8128 16.8943L9.64643 11.0607L15.8336 17.2478Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
            <div
              class="euknxbk-el-card-context-menu-item"
            >
              <span
                class="ej9yrfo-el-icon"
                style="font-size: 1.25rem;"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1.05582 19.5H1.18758L7.54482 18.1165C7.66865 18.0959 7.78339 18.0385 7.87418 17.9518L18.6452 7.1808C19.432 6.39567 19.871 5.32777 19.8639 4.21632C19.8654 3.09622 19.4278 2.02019 18.6452 1.21886C17.8601 0.432032 16.7922 -0.00696545 15.6807 0.000158881C14.5621 -0.00973953 13.4892 0.443255 12.7162 1.25184L1.97807 12.0228C1.88337 12.099 1.81434 12.2025 1.78042 12.3193L0.397035 18.6765C0.35927 18.9047 0.432657 19.1373 0.594688 19.3024C0.716306 19.4265 0.882058 19.4975 1.05582 19.5ZM15.6807 1.31772C17.2634 1.31759 18.5464 2.60054 18.5466 4.18322C18.5466 4.19425 18.5465 4.20528 18.5464 4.21632C18.5572 4.9703 18.2597 5.69603 17.7229 6.22557L13.6714 2.14112C14.2052 1.6107 14.9282 1.31438 15.6807 1.31772ZM12.7491 3.09641L16.8006 7.14788L7.41299 16.5025L3.36152 12.4839L12.7491 3.09641ZM2.8345 13.8016L6.09543 17.0625L1.91219 17.9848L2.8345 13.8016Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
            <div
              class="euknxbk-el-card-context-menu-item"
            >
              <span
                class="ej9yrfo-el-icon"
                style="font-size: 1.25rem;"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.5625 2.5H13.125V1.875C13.125 0.841133 12.2839 0 11.25 0H8.75C7.71613 0 6.875 0.841133 6.875 1.875V2.5H3.4375C2.57594 2.5 1.875 3.20094 1.875 4.0625V6.25C1.875 6.59516 2.15484 6.875 2.5 6.875H2.84156L3.38152 18.2142C3.42922 19.2156 4.25187 20 5.25437 20H14.7456C15.7482 20 16.5708 19.2156 16.6185 18.2142L17.1584 6.875H17.5C17.8452 6.875 18.125 6.59516 18.125 6.25V4.0625C18.125 3.20094 17.4241 2.5 16.5625 2.5ZM8.125 1.875C8.125 1.53039 8.40539 1.25 8.75 1.25H11.25C11.5946 1.25 11.875 1.53039 11.875 1.875V2.5H8.125V1.875ZM3.125 4.0625C3.125 3.8902 3.2652 3.75 3.4375 3.75H16.5625C16.7348 3.75 16.875 3.8902 16.875 4.0625V5.625C16.6824 5.625 3.92316 5.625 3.125 5.625V4.0625ZM15.3699 18.1547C15.354 18.4885 15.0798 18.75 14.7456 18.75H5.25437C4.9202 18.75 4.64598 18.4885 4.63012 18.1547L4.09297 6.875H15.907L15.3699 18.1547Z"
                    fill="currentColor"
                  />
                  <path
                    d="M10 17.5C10.3452 17.5 10.625 17.2202 10.625 16.875V8.75C10.625 8.40484 10.3452 8.125 10 8.125C9.65484 8.125 9.375 8.40484 9.375 8.75V16.875C9.375 17.2202 9.6548 17.5 10 17.5Z"
                    fill="currentColor"
                  />
                  <path
                    d="M13.125 17.5C13.4702 17.5 13.75 17.2202 13.75 16.875V8.75C13.75 8.40484 13.4702 8.125 13.125 8.125C12.7798 8.125 12.5 8.40484 12.5 8.75V16.875C12.5 17.2202 12.7798 17.5 13.125 17.5Z"
                    fill="currentColor"
                  />
                  <path
                    d="M6.875 17.5C7.22016 17.5 7.5 17.2202 7.5 16.875V8.75C7.5 8.40484 7.22016 8.125 6.875 8.125C6.52984 8.125 6.25 8.40484 6.25 8.75V16.875C6.25 17.2202 6.5298 17.5 6.875 17.5Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="e1sw79q5-el-card-main-wrap"
        >
          <div
            class="e1rjwng7-el-card-heading-wrap"
          >
            <h5
              class="e1n8fwu7-el-card-heading"
            >
              will-test-dev-co-voracious-person
            </h5>
            <h6
              class="e160zim8-el-card-sub-heading"
            >
              Will Test DevCo
            </h6>
            <h6
              class="e1nfyxby-el-card-sub-heading-additional"
            >
              Integration
            </h6>
          </div>
          <button
            class="e17hzamr-el-mobile-toggle"
          >
            <span
              class="ej9yrfo-el-icon"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 12 12"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="em92f2t-el-card-body-wrap"
        >
          Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ecbqyq6-el-snack-holder"
    />
    <div
      class="e6p05ya-el-fade-in cbaewf6-card-cursor ertn4py-el-card-wrap"
    >
      <div
        class="ec7f1kt-el-card-context-menu-wrapper"
      >
        <div
          class="ewpy0cr-el-card-context-menu-toggle"
        >
          <span
            class="ej9yrfo-el-icon"
            style="font-size: 1.25rem;"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="19"
                cy="4"
                fill="currentColor"
                r="1"
              />
              <circle
                cx="19"
                cy="10"
                fill="currentColor"
                r="1"
              />
              <circle
                cx="19"
                cy="16"
                fill="currentColor"
                r="1"
              />
            </svg>
          </span>
        </div>
        <div
          class="eh85zp4-el-card-context-menu-items"
        >
          <div
            class="euknxbk-el-card-context-menu-item"
          >
            <span
              class="ej9yrfo-el-icon"
              style="font-size: 1.25rem;"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15.8336 17.2478C16.1265 17.5407 16.6014 17.5407 16.8943 17.2478C17.1872 16.955 17.1872 16.4801 16.8943 16.1872L10.7071 10L16.8943 3.81282C17.1872 3.51992 17.1872 3.04505 16.8943 2.75216C16.6014 2.45926 16.1265 2.45926 15.8336 2.75216L9.64643 8.93934L3.8128 3.10571C3.51991 2.81282 3.04503 2.81282 2.75214 3.10571C2.45925 3.3986 2.45925 3.87348 2.75214 4.16637L8.58577 10L2.75214 15.8336C2.45925 16.1265 2.45925 16.6014 2.75214 16.8943C3.04503 17.1872 3.51991 17.1872 3.8128 16.8943L9.64643 11.0607L15.8336 17.2478Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
          <div
            class="euknxbk-el-card-context-menu-item"
          >
            <span
              class="ej9yrfo-el-icon"
              style="font-size: 1.25rem;"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.05582 19.5H1.18758L7.54482 18.1165C7.66865 18.0959 7.78339 18.0385 7.87418 17.9518L18.6452 7.1808C19.432 6.39567 19.871 5.32777 19.8639 4.21632C19.8654 3.09622 19.4278 2.02019 18.6452 1.21886C17.8601 0.432032 16.7922 -0.00696545 15.6807 0.000158881C14.5621 -0.00973953 13.4892 0.443255 12.7162 1.25184L1.97807 12.0228C1.88337 12.099 1.81434 12.2025 1.78042 12.3193L0.397035 18.6765C0.35927 18.9047 0.432657 19.1373 0.594688 19.3024C0.716306 19.4265 0.882058 19.4975 1.05582 19.5ZM15.6807 1.31772C17.2634 1.31759 18.5464 2.60054 18.5466 4.18322C18.5466 4.19425 18.5465 4.20528 18.5464 4.21632C18.5572 4.9703 18.2597 5.69603 17.7229 6.22557L13.6714 2.14112C14.2052 1.6107 14.9282 1.31438 15.6807 1.31772ZM12.7491 3.09641L16.8006 7.14788L7.41299 16.5025L3.36152 12.4839L12.7491 3.09641ZM2.8345 13.8016L6.09543 17.0625L1.91219 17.9848L2.8345 13.8016Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="euknxbk-el-card-context-menu-item"
          >
            <span
              class="ej9yrfo-el-icon"
              style="font-size: 1.25rem;"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.5625 2.5H13.125V1.875C13.125 0.841133 12.2839 0 11.25 0H8.75C7.71613 0 6.875 0.841133 6.875 1.875V2.5H3.4375C2.57594 2.5 1.875 3.20094 1.875 4.0625V6.25C1.875 6.59516 2.15484 6.875 2.5 6.875H2.84156L3.38152 18.2142C3.42922 19.2156 4.25187 20 5.25437 20H14.7456C15.7482 20 16.5708 19.2156 16.6185 18.2142L17.1584 6.875H17.5C17.8452 6.875 18.125 6.59516 18.125 6.25V4.0625C18.125 3.20094 17.4241 2.5 16.5625 2.5ZM8.125 1.875C8.125 1.53039 8.40539 1.25 8.75 1.25H11.25C11.5946 1.25 11.875 1.53039 11.875 1.875V2.5H8.125V1.875ZM3.125 4.0625C3.125 3.8902 3.2652 3.75 3.4375 3.75H16.5625C16.7348 3.75 16.875 3.8902 16.875 4.0625V5.625C16.6824 5.625 3.92316 5.625 3.125 5.625V4.0625ZM15.3699 18.1547C15.354 18.4885 15.0798 18.75 14.7456 18.75H5.25437C4.9202 18.75 4.64598 18.4885 4.63012 18.1547L4.09297 6.875H15.907L15.3699 18.1547Z"
                  fill="currentColor"
                />
                <path
                  d="M10 17.5C10.3452 17.5 10.625 17.2202 10.625 16.875V8.75C10.625 8.40484 10.3452 8.125 10 8.125C9.65484 8.125 9.375 8.40484 9.375 8.75V16.875C9.375 17.2202 9.6548 17.5 10 17.5Z"
                  fill="currentColor"
                />
                <path
                  d="M13.125 17.5C13.4702 17.5 13.75 17.2202 13.75 16.875V8.75C13.75 8.40484 13.4702 8.125 13.125 8.125C12.7798 8.125 12.5 8.40484 12.5 8.75V16.875C12.5 17.2202 12.7798 17.5 13.125 17.5Z"
                  fill="currentColor"
                />
                <path
                  d="M6.875 17.5C7.22016 17.5 7.5 17.2202 7.5 16.875V8.75C7.5 8.40484 7.22016 8.125 6.875 8.125C6.52984 8.125 6.25 8.40484 6.25 8.75V16.875C6.25 17.2202 6.5298 17.5 6.875 17.5Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="e1sw79q5-el-card-main-wrap"
      >
        <div
          class="e1rjwng7-el-card-heading-wrap"
        >
          <h5
            class="e1n8fwu7-el-card-heading"
          >
            will-test-dev-co-voracious-person
          </h5>
          <h6
            class="e160zim8-el-card-sub-heading"
          >
            Will Test DevCo
          </h6>
          <h6
            class="e1nfyxby-el-card-sub-heading-additional"
          >
            Integration
          </h6>
        </div>
        <button
          class="e17hzamr-el-mobile-toggle"
        >
          <span
            class="ej9yrfo-el-icon"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 12 12"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="em92f2t-el-card-body-wrap"
      >
        Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
